sim_pipeline/sim_lsst_galaxy_galaxy_lens_simulation.py

```diff
@@ -0,0 +1,283 @@
+    v_sigmamass = get_vd_from_stallermass(len_stellmass)
+    v_sigma = schechter_vdf(alpha=2.32, beta=2.67, vd_star=161, vd_min=200, vd_max=500) # Define a Schechter function with specific parameters to calculate velocity dispersion v_sigma
+    lensCos = LensCosmo(z_lens=redz_len, z_source=redz_source, cosmo=cosmolo) # Define a LensCosmo object with specific redshift values and cosmology
+#   theta_E = lensCos.sis_sigma_v2theta_E(v_sigma=v_sigma) # Calculate Einstein radius (theta_E) using lensCos and v_sigma
+    theta_E = lensCos.sis_sigma_v2theta_E(v_sigma=v_sigmamass)
+
+    # Define lensing models as dictionaries with specific parameters
+    kwargs_lens = [
+        {'theta_E': theta_E, 'e1': 0.2, 'e2': -0.1, 'center_x': 0, 'center_y': 0},  # SIE model
+        {'gamma1': 0.03, 'gamma2': 0.01, 'ra_0': 0, 'dec_0': 0}  # SHEAR model
+    ]
+    # turn magnitude kwargs into lenstronomy kwargs
+    kwargs_lens_light_g, kwargs_source_g, kwargs_ps_g = sim_g.magnitude2amplitude(kwargs_lens_light[0], kwargs_source[0])
+    kwargs_lens_light_r, kwargs_source_r, kwargs_ps_r = sim_r.magnitude2amplitude(kwargs_lens_light[1], kwargs_source[1])
+    kwargs_lens_light_i, kwargs_source_i, kwargs_ps_i = sim_i.magnitude2amplitude(kwargs_lens_light[2], kwargs_source[2])
+
+    # Generate lensed images for each band (g, r, i)
+    image_g = imSim_g.image(kwargs_lens, kwargs_source_g, kwargs_lens_light_g, kwargs_ps_g)
+    image_r = imSim_r.image(kwargs_lens, kwargs_source_r, kwargs_lens_light_r, kwargs_ps_r)
+    image_i = imSim_i.image(kwargs_lens, kwargs_source_i, kwargs_lens_light_i, kwargs_ps_i)
+
+    # add noise
+    image_g += sim_g.noise_for_model(model=image_g)
+    image_r += sim_r.noise_for_model(model=image_r)
+    image_i += sim_i.noise_for_model(model=image_i)
+
+    # use color with astropy
+    image = make_lupton_rgb(image_i, image_r, image_g, stretch=0.5)
+
+    # and plot it
+    ax.imshow(image, aspect='equal', origin='lower')
+    ax.get_xaxis().set_visible(False)
+    ax.get_yaxis().set_visible(False)
+    ax.autoscale(False)
+
+    print(v_sigmamass)
+    return v_sigma,theta_E,redz_source,redz_len
+
+n_horizont, n_vertical = 5, 3
+f, axes = plt.subplots(n_vertical, n_horizont, figsize=(n_horizont * 3, n_vertical * 3))
+sourcez_values = [] # First, create a list of z values
+lensz_values = []
+v_sigma_theta_Evalues = {'list1':[],'list2':[]}
+
+for i in range(n_horizont):  # plt image
+    for j in range(n_vertical):
+        process_image(ax=axes[j, i],sim_i=sim_i,sim_g=sim_g,sim_r=sim_r,cosmolo=cosmo)
+                      # add z to the list
+        vsigma, theE, zsource, zlens= process_image(ax=axes[j, i],sim_i=sim_i,sim_g=sim_g,sim_r=sim_r,cosmolo=cosmo)
+        sourcez_values.append(zsource)
+        lensz_values.append(zlens)
+        print(theE)
+
+        # extract the values of v_sigma and theta_E
+        v_sigma_theta_Evalues['list1'].append(vsigma)
+        v_sigma_theta_Evalues['list2'].append(theE)
+
+f.tight_layout()
```

Comment:
I would put these plotting routines in a different python file with is only dealing with plotting